evalml/pipelines/components/component_base.py

```diff
@@ -239,3 +243,33 @@ def update_parameters(self, update_dict, reset_fit=True):
         self._parameters.update(update_dict)
         if reset_fit:
             self._is_fitted = False
+
+    def _handle_nullable_types(self, X=None, y=None):
+        """Transforms X and y to remove any incompatible nullable types according to a component's needs.
+
+        Args:
+            X (pd.DataFrame, optional): Input data to a component of shape [n_samples, n_features].
+            y (pd.Series): The target of length [n_samples]. May contain nullable types.
+
```

Comment:
Copypasta nit: I don't think the `optional` only in the X is correct here, and I don't think y is the only argument that can contain nullables!